src/adapters/scene_direction.js

```diff
@@ -169,29 +170,55 @@ export default class SceneDirection {
     this.routes = [];
   }
 
-  addRouteFeature(route, vehicle) {
-    const layerStyle = getRouteStyle(vehicle, route.isActive);
-    layerStyle.id = `route_${route.id}`;
-    layerStyle.source = `source_${route.id}`;
-
-    const sourceId = `source_${route.id}`;
-    const sourceJSON = {
-      type: 'geojson',
-      data: normalizeToFeatureCollection(route.geometry),
-    };
-    this.map.addSource(sourceId, sourceJSON);
-    this.map.addLayer(layerStyle, map.routes_layer);
-
-    this.map.on('click', `route_${route.id}`, () => {
-      fire('select_road_map', route.id);
-    });
-
-    this.map.on('mouseenter', `route_${route.id}`, () => {
-      this.map.getCanvas().style.cursor = 'pointer';
-    });
+  getDataSources(route) {
+    const featureCollection = normalizeToFeatureCollection(route.geometry);
+    const sources = [];
+    if (this.vehicle === 'publicTransport') {
+      const walkFeatures = [], nonWalkFeatures = [];
```

Comment:
This is a lot of lines for something so simple and common (separate elements of an array in two arrays depending on a criterion). I'd love to introduce Ramda for this kind of stuff [R.partition](https://ramdajs.com/docs/#partition)
But let's discuss it later :)